[Clang][AMDGPU] Improve error message when device libraries for COV6 are missing

shiltian opened the pull request:
#130963 switches the default to COV6, which requires ROCm 6.3. Currently, if the
device libraries for COV6 are not found, the error message is not very helpful.
This PR provides a more informative error message in such cases.

shiltian commented:
* **#134745** <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/134745?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/llvm/llvm-project/134745?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

jhuber6 commented on clang/include/clang/Basic/DiagnosticDriverKinds.td:
```diff
@@ -67,7 +67,8 @@ def err_drv_no_cuda_libdevice : Error<
 def err_drv_no_rocm_device_lib : Error<
-  "cannot find ROCm device library%select{| for %1| for ABI version %1}0; provide its path via "
+  "cannot find ROCm device library%select{| for %1| for ABI version %1"
+  "%select{|, which requires ROCm 6.3 or higher}2}0; provide its path via "
```
Would it make more sense to put this as a variable and calculate it in the ROCMInstallationDetector?

shiltian replied:
which part? the rocm version?

jhuber6 replied:
Yes.

shiltian replied:
So this ROCm 6.3 thingy could be updated and extended easily in the near future by `%select{6.3|7.0|...}3`, but for now we just don't have a use case here. That's why I "hard coded" it to 6.3 at the moment. I guess when we have COV7 by default, we can do the part.

arsenm replied:
Best to do it now or it will never happen, and will just print this outdated message

shiltian committed:
[Clang][AMDGPU] Improve error message when device libraries for COV6 are missing

130963 switches the default to COV6, which requires ROCm 6.3. Currently, if the
device libraries for COV6 are not found, the error message is not very helpful.
This PR provides a more informative error message in such cases.
